pkg/proxy/userspace/proxier_test.go

```diff
@@ -651,6 +651,8 @@ func TestTCPProxyUpdateDeleteUpdate(t *testing.T) {
 	}
 	waitForNumProxyLoops(t, p, 0)
 
+	p.syncProxyRules() // ensure the endpoint has been deleted
```

Comment:
I don't think this correct. Waiting for syncProxyRules() to run is exactly what waitForNumProxyLoops() is supposed to do. This should be effectively running syncProxyRules() twice when we expect to only need it run once.

So the question is either:

1. is waitForNumProxyLoops() doing the right thing and actually waiting for syncProxyRules() to run once,
2. is syncProxyRules() getting correctly triggered by the OnServiceDelete

Reply:
@knight42 see https://github.com/kubernetes/kubernetes/pull/93979 for a larger cleanup that hopefully addresses the flake too

Reply:
@dcbw Great! Thanks for your fix!